File: daily_problems/2025/09/0918/personal_submission/cf105493c_yawn_sean.cpp
#include <bits/stdc++.h>
#pragma GCC optimize("O3,Ofast,unroll-loops")
// #include "atcoder/all"

using namespace std;
auto rng = mt19937(random_device()());
auto rngl = mt19937_64(random_device()());

// Let's set a bit and flow!
// I came, I divided, I conquered!
// using namespace __gnu_cxx;
// using namespace __gnu_pbds;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    int n;
    cin >> n;

    int x, y;
    cin >> x >> y;

    vector<pair<int, int>> pts;

    for (int i = 0; i < n - 1; i ++) {
        int nx, ny;
        cin >> nx >> ny;

        pts.emplace_back(nx - x, ny - y);
        x = nx, y = ny;
    }

    if (n == 1) {
        cout << "YES\n";
        cout << "1 0\n";
        cout << "0 1\n";
        return 0;
    }

    stable_sort(pts.begin(), pts.end(), [&] (pair<int, int> x, pair<int, int> y) {
        return 1ll * x.first * y.second < 1ll * x.second * y.first;
    });

    auto [x1, y1] = *pts.begin();
    auto [x2, y2] = *pts.rbegin();

    int vx, vy;

    if (1ll * x1 * x2 + 1ll * y1 * y2 >= 0)
        vx = x1 + x2, vy = y1 + y2;
    else {
        int vx1 = y1, vy1 = -x1;
        if (1ll * vx1 * x2 + 1ll * vy1 * y2 < 0) vx1 = -vx1, vy1 = -vy1;
        int vx2 = y2, vy2 = -x2;
        if (1ll * vx2 * x1 + 1ll * vy2 * y1 < 0) vx2 = -vx2, vy2 = -vy2;
        vx = vx1 + vx2, vy = vy1 + vy2;
    }

    for (auto &[x, y]: pts) {
        if (1ll * vx * x + 1ll * vy * y <= 0) {
            return cout << "NO", 0;
        }
    }

    cout << "YES\n";
    cout << vx << ' ' << vy << '\n';
    cout << -vy << ' ' << vx << '\n';

    return 0;
}

File: daily_problems/2025/09/0918/personal_submission/cf105493c_yawn_sean.py
# Submission link: https://codeforces.com/gym/105493/submission/339027375
def main():
    n = II()
    x, y = MII()
    
    if n == 1:
        print('YES')
        print(1, 0)
        print(0, 1)
        exit()
    
    xs = []
    ys = []
    
    for _ in range(n - 1):
        nx, ny = MII()
        xs.append(nx - x)
        ys.append(ny - y)
        x, y = nx, ny
    
    def cmp(i, j):
        xi, yi = xs[i], ys[i]
        xj, yj = xs[j], ys[j]
    
        v = xi * yj - yi * xj
        if v > 0: return 1
        if v < 0: return -1
        return 0
    
    st_range = sorted(range(n - 1), key=cmp_to_key(cmp))
    
    i1 = st_range[0]
    i2 = st_range[-1]
    
    x1, y1 = xs[i1], ys[i1]
    x2, y2 = xs[i2], ys[i2]
    
    if x1 * x2 + y1 * y2 >= 0:
        vx, vy = x1 + x2, y1 + y2
    else:
        vx1, vy1 = y1, -x1
        if vx1 * x2 + vy1 * y2 < 0: vx1, vy1 = -vx1, -vy1
        vx2, vy2 = y2, -x2
        if vx2 * x1 + vy2 * y1 < 0: vx2, vy2 = -vx2, -vy2
        vx, vy = vx1 + vx2, vy1 + vy2
    
    for i in range(n - 1):
        if vx * xs[i] + vy * ys[i] <= 0:
            exit(print('NO'))
    
    print('YES')
    print(vx, vy)
    print(-vy, vx)